hibernate-core/src/main/java/org/hibernate/dialect/AbstractTransactSQLDialect.java

```diff
@@ -86,10 +86,10 @@ public void initializeFunctionRegistry(QueryEngine queryEngine) {
 				.setExactArgumentCount( 1 )
 				.register();
 
-		queryEngine.getSqmFunctionRegistry().patternDescriptorBuilder( "(?1 % ?2)" )
+		queryEngine.getSqmFunctionRegistry().patternDescriptorBuilder( "mod", "(?1 % ?2)" )
```

Comment:
I *really* dislike this form which is why I changed it.  Please revert these types of changes.

Reply:
@sebersole I want the functions to know their own name. Because I want them to have a much stronger "identity" about them for stuff like pretty printing, etc.